sql-infra: add resource control feature (#12353)
pingcap/docs: system-variables.md

</CustomContent>

### `tidb_enable_resource_control` <span class="version-mark">New in v6.6.0</span>

> **Warning:**
>

> > This feature is experimental and its form and usage may change in subsequent versions.

Reviewer: ```suggestion
> This feature is experimental and its form and usage might change in subsequent versions.
```

- Scope: GLOBAL
- Persists to cluster: Yes
- Default value: `OFF`
- Type: Boolean

> - This variable is a switch for [the resource control feature](/tidb-resource-control.md). When this variable is set to `ON`, the cluster supports resource isolation of applications by resource group.

Author: ```suggestion
- This variable is a switch for [the resource control feature](/tidb-resource-control.md). When this variable is set to `ON`, the TiDB cluster supports resource isolation of applications based on resource groups.
```

> - This variable is a switch for [the resource control feature](/tidb-resource-control.md). When this variable is set to `ON`, the TiDB cluster supports resource isolation of applications based on resource groups.

Author: ```suggestion
- This variable is a switch for [the resource control feature](/tidb-resource-control.md). When this variable is set to `ON`, the TiDB cluster can isolate application resources based on resource groups.
```
